test_helpers/main.go

```diff
@@ -0,0 +1,238 @@
+	RetryTimeout time.Duration
+}
+
+// TarantoolInstance is a data for instance graceful shutdown and cleanup.
+type TarantoolInstance struct {
+	// Cmd is a Tarantool command. Used to kill Tarantool process.
+	Cmd *exec.Cmd
+
+	// WorkDir is a directory with tarantool data. Cleaned up after run.
+	WorkDir string
+}
+
+func isReady(server string, opts *tarantool.Opts) error {
+	var err error
+	var conn *tarantool.Connection
+	var resp *tarantool.Response
+
+	conn, err = tarantool.Connect(server, *opts)
+	if err != nil {
+		return err
+	}
+	if conn == nil {
+		return errors.New("Conn is nil after connect")
+	}
+	defer conn.Close()
+
+	resp, err = conn.Ping()
+	if err != nil {
+		return err
+	}
+	if resp == nil {
+		return errors.New("Response is nil after ping")
+	}
+
+	return nil
+}
+
+var (
+	// Used to extract Tarantool version (major.minor.patch).
+	tarantoolVersionRegexp *regexp.Regexp
+)
+
+func init() {
+	tarantoolVersionRegexp = regexp.MustCompile(`Tarantool (?:Enterprise )?(\d+)\.(\d+)\.(\d+).*`)
+}
+
+// atoiUint64 parses string to uint64.
+func atoiUint64(str string) (uint64, error) {
+	res, err := strconv.ParseUint(str, 10, 32)
+	if err != nil {
+		return 0, fmt.Errorf("cast to number error (%s)", err)
+	}
+	return res, nil
+}
+
+// IsTarantoolVersionLess checks if tarantool version is less
+// than passed <major.minor.patch>. Returns error if failed
+// to extract version.
+func IsTarantoolVersionLess(majorMin uint64, minorMin uint64, patchMin uint64) (bool, error) {
+	var major, minor, patch uint64
+
+	out, err := exec.Command("tarantool", "--version").Output()
+
+	if err != nil {
+		return true, err
+	}
+
+	parsed := tarantoolVersionRegexp.FindStringSubmatch(string(out))
+
+	if parsed == nil {
+		return true, errors.New("regexp parse failed")
+	}
+
+	if major, err = atoiUint64(parsed[1]); err != nil {
+		return true, fmt.Errorf("failed to parse major: %s", err)
+	}
+
+	if minor, err = atoiUint64(parsed[2]); err != nil {
+		return true, fmt.Errorf("failed to parse minor: %s", err)
+	}
+
+	if patch, err = atoiUint64(parsed[3]); err != nil {
+		return true, fmt.Errorf("failed to parse patch: %s", err)
+	}
+
+	if major < majorMin {
+		return true, nil
+	}
+
+	if major == majorMin && minor < minorMin {
+		return true, nil
+	}
+
+	if major == majorMin && minor == minorMin && patch < patchMin {
+		return true, nil
+	}
+
+	return false, nil
+}
+
+// StartTarantool starts a tarantool instance for tests
+// with specifies parameters (refer to StartOpts).
+// Process must be stopped with StopTarantool.
+func StartTarantool(startOpts StartOpts) (TarantoolInstance, error) {
+	// Prepare tarantool command.
+	var inst TarantoolInstance
+	inst.Cmd = exec.Command("tarantool", startOpts.InitScript)
+
+	inst.Cmd.Env = append(
+		os.Environ(),
+		fmt.Sprintf("TEST_TNT_WORK_DIR=%s", startOpts.WorkDir),
+		fmt.Sprintf("TEST_TNT_LISTEN=%s", startOpts.Listen),
+	)
+
+	// Clean up existing work_dir.
+	err := os.RemoveAll(startOpts.WorkDir)
```

Comment:
You are using cleanup in several places. I suggest you move it to a separate helper.

Reply:
I don't see a reason to make
```go
	err := os.RemoveAll(startOpts.WorkDir)
	if err != nil {
		return inst, err
	}
```
script into a
```go
	err := cleanDir(startOpts.WorkDir)
	if err != nil {
		return inst, err
	}
```